File: ruby/decorator/StarbuzzCoffee/decorator/condiments/milk.rb
require './decorator/condiment_decorator'

class Milk < CondimentDecorator

  def initialize(beverage)
    @@beverage = beverage
  end

  def get_description
    super
    @@beverage.get_description + ", Milk"
  end

  def cost
    super
    @@beverage.cost + 0.89
  end

  private
  @@beverage = Beverage.new

end

File: python/decorator/StarbuzzCoffee/decorator/condiments/Milk.py
from component.Beverage import Beverage
from decorator.CondimentDecorator import CondimentDecorator


class Milk(CondimentDecorator):
    __beverage = Beverage()

    def __init__(self, beverage):
        super().__init__()
        self.__beverage = beverage

    def get_description(self):
        return self.__beverage.get_description() + ", Milk"

    def cost(self):
        return self.__beverage.cost() + 0.99
